src/core/matchers/matchersUtil.js

```diff
@@ -193,9 +193,9 @@ getJasmineRequireObj().matchersUtil = function(j$) {
       return false;
     }
 
-    var aIsPromise = j$.isPromise(a);
```

Comment:
I don't think we want to move Jasmine's equality away from doing an explicit constructor check at this point. It is possible for non-Promises to have a `then` function available, and we don't want to break deep equality for anyone relying on that functionality.